core/src/main/java/com/linecorp/armeria/common/sse/ServerSentEvent.java

```diff
@@ -0,0 +1,109 @@
+/*
+ * Copyright 2019 LINE Corporation
+ *
+ * LINE Corporation licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+package com.linecorp.armeria.common.sse;
+
+import java.time.Duration;
+import java.util.stream.Stream;
+
+import javax.annotation.Nullable;
+
+import org.reactivestreams.Publisher;
+
+import com.linecorp.armeria.server.annotation.ProducesEventStream;
+import com.linecorp.armeria.server.annotation.ServerSentEventResponseConverterFunction;
+
+/**
+ * An interface for the <a href="https://www.w3.org/TR/eventsource/">Server-sent Events</a> specification.
+ * If a {@link Publisher} or {@link Stream} produces objects which implement this interface, it can be
+ * converted into a text event stream by {@link ServerSentEventResponseConverterFunction}.
+ *
+ * @see ProducesEventStream
+ * @see ServerSentEventResponseConverterFunction
+ */
+public interface ServerSentEvent {
+
+    /**
+     * Creates a new empty {@link ServerSentEvent}.
+     */
+    static ServerSentEvent ofEmpty() {
```

Comment:
Curious if we should rename this to `empty()` after making it return a singleton. We have `empty()` in other classes already, so it may be better renaming for consistency.